jwst/pipeline/calwebb_spec3.py

```diff
@@ -228,15 +242,17 @@ def process(self, input):
             if exptype in ['MIR_MRS']:
                 result = self.mrs_imatch(result)
 
-            # Call outlier detection
+            # Call outlier detection and pixel replacement
             if exptype not in SLITLESS_TYPES:
                 # Update the asn table name to the level 3 instance so that
                 # the downstream products have the correct table name since
                 # the _cal files are not saved they will not be updated
                 for cal_array in result:
                     cal_array.meta.asn.table_name = op.basename(input_models.asn_table_name)
                 result = self.outlier_detection(result)
-
+                # interpolate pixels are which have a NaN value or are flagged
+                # as DO_NOT_USE or NON_SCIENCE
+                result = self.pixel_replace(result)
                 # Resample time. Dependent on whether the data is IFU or not.
                 resample_complete = None
```

Comment:
I think we should be calling `pixel_replace` here before either `cube_build` or `resample_spec` is called, to give folks the option of applying it.